lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,38 @@
+require 'forwardable'
+
+module Semian
+  class AtomicEnum #:nodoc:
+    extend Forwardable
+
+    def_delegators :@integer, :destroy
+
+    def initialize(symbol_list)
+      @integer = Semian::AtomicInteger.new
+      initialize_lookup(symbol_list)
+    end
+
+    def increment(val = 1)
+      @integer.value = (@integer.value + val) % @sym_to_num.size
```

Comment:
It's what I'm trying to say. Even if your `AtomicInteger` was actually atomic, the way to you use it is not.